catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreAutoPublisher.java

```diff
@@ -0,0 +1,146 @@
+
+        if (reference != null && bundleContext != null) {
+            RegistryStore registryStore = (RegistryStore) bundleContext.getService(reference);
+
+            Boolean previousState = registryStoreMap.put(registryStore.getConfigurationPid(), registryStore.isAutoPush());
+
+            SourceMonitor storeRegisteredSourceMonitor = new SourceMonitor() {
+                @Override
+                public void setAvailable() {
+                    if (registryStore.isAutoPush()) {
+                        registryPublish(registryStore);
+                    }
+                }
+
+                @Override
+                public void setUnavailable() {
+                }
+            };
+
+            if (registryStore.isAvailable(storeRegisteredSourceMonitor) && registryStore.isAutoPush()) {
+                if(previousState == null || !previousState) {
+                    registryPublish(registryStore);
+                }
+            }
+        }
+    }
+
+    void registryPublish(RegistryStore registryStore) {
+
+        if (!registryStore.getRegistryId()
+                .isEmpty()) {
+            try {
+
+                Optional<Metacard> registryIdentityMetacardOpt =
+                        Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
+
+                if (registryIdentityMetacardOpt.isPresent()) {
+                    Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
+                    String localRegistryId = registryIdentityMetacard.getAttribute(
+                            RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString();
+                    Security.runAsAdminWithException(() -> {
+                        registryPublicationService.publish(localRegistryId,
+                                registryStore.getRegistryId());
+                        return null;
+                    });
+                }
+            } catch (Exception e) {
+                LOGGER.error("Failed to auto-push registry configuration from {}, {}",
+                        registryStore.getRegistryId(),
+                        e);
+            }
+        }
+    }
+
+    BundleContext getBundleContext() {
+        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+        if (bundle != null) {
+            return bundle.getBundleContext();
+        }
+        return null;
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        if (((ServiceEvent) event.getProperty(EventConstants.EVENT)).getType()
+                != ServiceEvent.MODIFIED) {
+            return;
+        }
+        String pid = event.getProperty(Constants.SERVICE_PID)
+                .toString();
+
+        BundleContext bundleContext = getBundleContext();
+
+        RegistryStore registryStore =
+                (RegistryStore) bundleContext.getService(((ServiceEvent) event.getProperty(
```

Comment:
While it's highly unlikely it'd be null here, you're not protecting against an unset `bundleContext`.